src/main/java/usi/si/seart/service/StaticCodeAnalysisServiceImpl.java

```diff
@@ -73,28 +73,18 @@ public Future<Set<GitRepoMetric>> getCodeMetrics(@NotNull GitRepo repo, boolean
         return new AsyncResult<>(metrics);
     }
 
-
-    public Future<Set<GitRepoMetric>> getCodeMetrics(@NotNull String repo_name, boolean persist) throws StaticCodeAnalysisException {
-        GitRepo repo = gitRepoRepository.findGitRepoByName(repo_name).orElse(null);
-        if (repo == null)
-            log.warn("Computing metrics for '{}' but did NOT find repo in the database.", repo_name);
-
-        return getCodeMetrics(repo, persist);
-
-    }
-
     private Set<GitRepoMetric> convert(@Nullable GitRepo repo, @NotNull JsonObject source) {
         return source.entrySet().stream().filter((Map.Entry<String, JsonElement> entry) ->
                 !entry.getKey().equals("header") && !entry.getKey().equals("SUM")
         ).map(entry -> {
             JsonObject stat = entry.getValue().getAsJsonObject();
             GitRepoMetric.GitRepoMetricBuilder builder = GitRepoMetric.builder();
 
-            MetricLanguage language = MetricLanguage.builder().language(entry.getKey()).build();
+            MetricLanguage language = metricLanguageService.getOrCreateMetricLanguage(entry.getKey());
             builder.language(language);
             if (repo != null) {
                 builder.repo(repo);
-                builder.id(new GitRepoMetricKey(repo.getId(), language.getLanguage()));
+                builder.id(new GitRepoMetricKey(repo.getId(), language.getId()));
             }
```

Comment:
This is the only place in the method where you use `repo`. Perhaps it's better to move it outside, and place the code after the `convert` call? Then we could hypothetically extract the conversion logic into a converter, but we would use array types instead of generics. What I mean is that if you define a converter that maps from an array of `JsonObject` to `MetricLanguage`, you can define it as:
```java
public class JsonObjectsToMetricLanguagesConverter implements Converter<JsonObject[], MetricLanguage[]> {

    @Override
    @NonNull
    public Label[] convert(@NonNull JsonObject[] source) {
        // Convert using a combination of streaming, mapping and collection to array...
    }
}
```

I'm also sure `JsonArray` can be used equivalently.
Then just use the conversion service in your code as follows:

```java
JsonObject[] jsonObjects = ... ;
Collection<MetricLanguage> metrics = List.of(
	conversionService.convert(jsonObjects, MetricLanguage[].class)
);
```

Reply:
Yes using a Spring converter class was my initial intention, but since I was using composite types and seeing how complicated it was to make use of the converter class, I made my own method.

Ill look into it, thanks.